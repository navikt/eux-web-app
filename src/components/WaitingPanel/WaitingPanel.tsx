import classNames from 'classnames'
import { HorizontalSeparatorDiv } from '@navikt/hoykontrast'
import { BodyLong, Loader } from '@navikt/ds-react'
import PT from 'prop-types'
import React from 'react'
import styled from 'styled-components'

export const WaitingPanelDiv = styled.div`
  display: flex;
  align-items: center;
  flex-direction: column;
  &.rowDirection {
     flex-direction: row;
  }
  .oneLine {
    display: inline-block;
    vertical-align: top;
  }
`

export type WaitingPanelSize = 'xsmall'| 'small' | 'medium'| 'large' | 'xlarge'| '2xlarge'

export interface WaitingPanelProps {
  className?: string
  size?: WaitingPanelSize
  style?: React.CSSProperties
  message?: string,
  oneLine?: boolean
}

const WaitingPanel: React.FC<WaitingPanelProps> = ({
  className, size = 'medium', style = {}, message = 'Vennligst vent...', oneLine = false
}: WaitingPanelProps): JSX.Element | null => (
  <WaitingPanelDiv
    style={style}
    className={classNames(className, { rowDirection: oneLine })}
  >
    <Loader type={size} />
    {message && (
      <>
        <HorizontalSeparatorDiv />
        <BodyLong
          className={classNames({ oneLine: oneLine })}
          data-test-id='c-waitingpanel__text-id'
        >
          {message}
        </BodyLong>
      </>
    )}
  </WaitingPanelDiv>
)

WaitingPanel.propTypes = {
  className: PT.string,
  message: PT.string,
  oneLine: PT.bool,
  size: PT.oneOf(['xsmall', 'small', 'medium', 'large', 'xlarge', '2xlarge']),
  style: PT.object
}
WaitingPanel.displayName = 'WaitingPanel'
export default WaitingPanel
