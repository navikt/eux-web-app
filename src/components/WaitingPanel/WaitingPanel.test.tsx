import { render } from '@testing-library/react'
import React from 'react'
import WaitingPanel from './WaitingPanel'

describe('components/WaitingPanel', () => {
  let wrapper: any

  it('Renders', () => {
    wrapper = render(<WaitingPanel message='' />)
    expect(wrapper.isEmptyRender()).toBeFalsy()
    expect(wrapper).toMatchSnapshot()
  })

  it('Has proper HTML structure', () => {
    wrapper = render(<WaitingPanel message='testmessage' />)
    expect(wrapper.exists('.c-waitingPanel')).toBeTruthy()
    expect(wrapper.find('.c-waitingPanel__message').hostNodes().text()).toEqual('testmessage')
  })
})
