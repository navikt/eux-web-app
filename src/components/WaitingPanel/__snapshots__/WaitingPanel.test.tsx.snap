// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/WaitingPanel Renders 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0.rowDirection {
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c0 .oneLine {
  display: inline-block;
  vertical-align: top;
}

<WaitingPanel
  message=""
>
  <styled.div
    className=""
    style={Object {}}
  >
    <div
      className="c0"
      style={Object {}}
    >
      <NavFrontendSpinner
        aria-label="Laster innhold"
        transparent={false}
        type="M"
      >
        <Icon
          aria-label="Laster innhold"
          className="spinner spinner--m"
          kind="spinner-negativ"
          size={32}
        >
          <svg
            aria-label="Laster innhold"
            className="spinner spinner--m"
            focusable="false"
            height={32}
            kind="spinner-negativ"
            preserveAspectRatio="xMidYMid"
            viewBox="0 0 50 50"
            width={32}
          >
            <title>
              Venter...
            </title>
            <circle
              cx="25"
              cy="25"
              fill="none"
              r="20"
              stroke="#eee"
              strokeWidth="5"
              xmlns="http://www.w3.org/2000/svg"
            />
            <circle
              cx="25"
              cy="25"
              fill="none"
              r="20"
              stroke="#888"
              strokeDasharray="50 155"
              strokeLinecap="round"
              strokeWidth="5"
            />
          </svg>
        </Icon>
      </NavFrontendSpinner>
    </div>
  </styled.div>
</WaitingPanel>
`;
